Dockerfile

```diff
@@ -1,45 +1,91 @@
 
-RUN apt-get update && apt-get install -y curl bzip2 gcc
+RUN apt-get update && apt-get install -y curl bzip2 \
+    build-essential unzip sqlite3 libsqlite3-dev
 
 # Miniconda - Python 3.6, 64-bit, x86, latest
 ARG CONDA_OS=Linux
-RUN curl -sL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o mconda-install.sh && \
-    bash -x mconda-install.sh -b -p miniconda && \
-    rm mconda-install.sh
-ENV PATH="/miniconda/bin:$PATH"
-
 ARG CONDA_ADD_PACKAGES=""
-RUN conda create -y -q -n sqlflow-dev python=3.6 ${CONDA_ADD_PACKAGES}
-
-RUN echo ". /miniconda/etc/profile.d/conda.sh" >> ~/.bashrc && \
-    echo "source activate sqlflow-dev" >> ~/.bashrc
-
 ARG PIP_ADD_PACKAGES=""
-RUN /bin/bash -c "source activate sqlflow-dev && python -m pip install \
+
+RUN curl -sL https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -o mconda-install.sh && \
+    bash -x mconda-install.sh -b -p miniconda && \
+    rm mconda-install.sh && \
+    /miniconda/bin/conda create -y -q -n sqlflow-dev python=3.6 ${CONDA_ADD_PACKAGES} && \
+    echo ". /miniconda/etc/profile.d/conda.sh" >> ~/.bashrc && \
+    echo "source activate sqlflow-dev" >> ~/.bashrc && \
+    /bin/bash -c "source /miniconda/bin/activate sqlflow-dev && python -m pip install \
     tensorflow==2.0.0-alpha0 \
     mysql-connector-python \
     impyla \
     jupyter \
     sqlflow \
     ${PIP_ADD_PACKAGES} \
     "
+ENV PATH="/miniconda/bin:$PATH"
+
+# Install go 1.11.5
+RUN apt-get install -y wget git && \
+    wget --quiet https://dl.google.com/go/go1.11.5.linux-amd64.tar.gz && \
+    tar -C /usr/local -xzf go1.11.5.linux-amd64.tar.gz && \
+    rm go1.11.5.linux-amd64.tar.gz && \
+    mkdir -p /go
+ENV GOPATH /go
+ENV PATH $PATH:/usr/local/go/bin:/go/bin
+
+RUN go get github.com/golang/protobuf/protoc-gen-go && \
+    mv /go/bin/protoc-gen-go /usr/local/bin/ && \
+    go get golang.org/x/lint/golint
+
+# Install protobuf compiler
+RUN wget --quiet https://github.com/protocolbuffers/protobuf/releases/download/v3.6.1/protoc-3.6.1-linux-x86_64.zip && \
+    unzip -qq protoc-3.6.1-linux-x86_64.zip -d /usr/local && \
+    rm protoc-3.6.1-linux-x86_64.zip
+
+# install mysql without a password prompt
+RUN echo 'mysql-server mysql-server/root_password password root' | debconf-set-selections && \
+    echo 'mysql-server mysql-server/root_password_again password root' | debconf-set-selections && \
+    apt-get install -y mysql-server && \
+    mkdir -p /var/run/mysqld && \
+    mkdir -p /var/lib/mysql && \
+    chown mysql:mysql /var/run/mysqld && \
+    mkdir -p /docker-entrypoint-initdb.d
+VOLUME /var/lib/mysql
+
+# Build SQLFlow binaries
+RUN mkdir -p /go/src/github.com/sql-machine-learning && \
+    cd /go/src/github.com/sql-machine-learning && \
+    git clone -q https://github.com/sql-machine-learning/sqlflow.git && \
+    cd sqlflow && \
+    go generate ./... && \
+    go get -v -t ./... && \
+    go install -v ./... && \
+    mv /go/bin/demo /usr/bin && \
```

Comment:
Since we have `ENV PATH $PATH:/usr/local/go/bin:/go/bin`, no need to move it to `/usr/bin`.

A developer may test his package using `go install ./... && demo ...`, this command updates the package at `/go/bin/demo` but not `/usr/bin/demo`. Since `/go/bin` comes after `/usr/bin`, the developer may get confused.

Removing `mv /go/bin/demo /usr/bin && mv /go/bin/sqlflowserver /usr/bin` will solve this issue.

Reply:
Done!